File: 卡码/代码随想录/1/06-二叉树/C++/06-二叉搜索树修改与构造/LeetCode_0108.cpp
//108. 将有序数组转换为二叉搜索树
#include <iostream>
#include <vector>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
	TreeNode* traversal(vector<int>& nums,int left,int right){
		if(left>right) return nullptr;
		int mid=(left+right)/2;
		TreeNode* root=new TreeNode(nums[mid]);
		root->left=traversal(nums,left,mid-1);
		root->right=traversal(nums,mid+1,right);
		return root;
	}
  TreeNode* sortedArrayToBST(vector<int>& nums) {
		return traversal(nums,0,nums.size()-1);
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/06-二叉树/Python/06-二叉搜索树修改与构造/LeetCode_0108.py
#108. 将有序数组转换为二叉搜索树
from typing import *
from ..TreeNode import *

class Solution:
  def traversal(self,nums:List[int],left:int,right:int)->Optional[TreeNode]:
    if left>right:
      return None
    mid:int=(left+right)//2
    root:TreeNode=TreeNode(nums[mid])
    root.left=self.traversal(nums,left,mid-1)
    root.right=self.traversal(nums,mid+1,right)
    return root
  def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
    return self.traversal(nums,0,len(nums)-1)
